File: vertx-elasticsearch/src/main/java/fr/myprysm/vertx/elasticsearch/action/search/NestedIdentity.java
package fr.myprysm.vertx.elasticsearch.action.search;

import io.vertx.codegen.annotations.DataObject;
import io.vertx.core.json.JsonObject;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.Accessors;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Accessors(chain = true)
@DataObject(generateConverter = true)
public class NestedIdentity {
    private String field;
    private int offset;
    private NestedIdentity child;

    public NestedIdentity(NestedIdentity other) {
        field = other.field;
        offset = other.offset;
        child = other.child;
    }

    /**
     * Build a new <code>NestedIdentity</code> from a <code>JsonObject</code>.
     *
     * @param json the <code>JsonObject</code>
     */
    public NestedIdentity(JsonObject json) {
        NestedIdentityConverter.fromJson(json, this);
    }

    /**
     * Transforms the <code>NestedIdentity</code> into a <code>JsonObject</code>.
     *
     * @return the <code>JsonObject</code>
     */
    public JsonObject toJson() {
        JsonObject json = new JsonObject();
        NestedIdentityConverter.toJson(this, json);
        return json;
    }
}

File: vertx-elasticsearch/src/main/kotlin/fr/myprysm/vertx/elasticsearch/kotlin/action/search/NestedIdentity.kt
package fr.myprysm.vertx.elasticsearch.kotlin.action.search

import fr.myprysm.vertx.elasticsearch.action.search.NestedIdentity

fun NestedIdentity(
  child: fr.myprysm.vertx.elasticsearch.action.search.NestedIdentity? = null,
  field: String? = null,
  offset: Int? = null): NestedIdentity = fr.myprysm.vertx.elasticsearch.action.search.NestedIdentity().apply {

  if (child != null) {
    this.setChild(child)
  }
  if (field != null) {
    this.setField(field)
  }
  if (offset != null) {
    this.setOffset(offset)
  }
}

